nemo_skills/dataset/prepare.py

```diff
@@ -29,6 +30,11 @@
         choices=["math", "code", "chat", "multichoice"],
         help='Can specify a dataset groups here',
     )
+    parser.add_argument(
```

Comment:
This should go inside lean-specific prepare.py not this general one that does it for all benchmarks. And then you can just call you dataset prepare.py directly and provide this option

Reply:
Since we want to be able to use auto formalization as a validation step, maybe it is better to live it for all the datasets? 
All my experiments are based on "math" dataset. Should I add this step directly to math dataset preparation?

Reply:
yeah, I guess you're right. Let's keep it here but how about we only do it for datasets from "math" group?